specutils/tests/test_continuum.py

```diff
@@ -59,3 +62,91 @@ def test_continuum_calculation():
+
+    # Smooth in the same way fit_generic_continuum does.
+    spectrum_smoothed = median_smooth(spectrum, 3)
+
+    # Check that a full width window recovers the original, non-windowed fit.
+    g1_fit = fit_continuum(spectrum_smoothed, window=(0.*u.um, 10.*u.um))
+
+    spectrum_normalized = spectrum / g1_fit(spectrum.spectral_axis)
+
+    y_continuum_fitted_expected = np.array([1.15139925, 0.98509363, 0.73700614, 1.00911864, 0.913129,
+                                            0.93145533, 0.94904202, 1.04162879, 0.90851397, 0.9494352,
+                                            1.07812394, 1.06376489, 0.98705237, 0.94569623, 0.83502377,
+                                            0.91909416, 0.89662208, 1.01458511, 0.96124191, 0.94847744])
+
+    assert np.allclose(spectrum_normalized.flux.value[::10], y_continuum_fitted_expected, atol=1e-5)
+
+
+def test_continuum_spectral_region():
+    """
+    As before, but with a SpectralRegion as window.
+    """
+
+    x_single_continuum, y_single_continuum = single_peak_continuum()
+    spectrum = Spectrum1D(flux=y_single_continuum*u.Jy, spectral_axis=x_single_continuum*u.um)
+
+    # Smooth in the same way fit_generic_continuum does.
+    spectrum_smoothed = median_smooth(spectrum, 3)
+
+    # Check that a full width window recovers the original, non-windowed fit.
+    region = SpectralRegion(0.*u.um, 10.*u.um)
+    g1_fit = fit_continuum(spectrum_smoothed, window=region)
+
+    spectrum_normalized = spectrum / g1_fit(spectrum.spectral_axis)
+
+    y_continuum_fitted_expected = np.array([1.15139925, 0.98509363, 0.73700614, 1.00911864, 0.913129,
+                                            0.93145533, 0.94904202, 1.04162879, 0.90851397, 0.9494352,
+                                            1.07812394, 1.06376489, 0.98705237, 0.94569623, 0.83502377,
+                                            0.91909416, 0.89662208, 1.01458511, 0.96124191, 0.94847744])
+
+    assert np.allclose(spectrum_normalized.flux.value[::10], y_continuum_fitted_expected, atol=1e-5)
+
+
+def test_continuum_window_no_noise():
+    """
+    This test repeats the setup from the previous tests, but uses windows to
+    select specific regions to fit. It uses a synthetic spectrum with no
+    noise component. That way, numerical effects coming from the model fit
+    itself can be more easily spotted.
+    """
+
+    x_single_continuum, y_single_continuum = single_peak_continuum(noise=0.)
+    spectrum = Spectrum1D(flux=y_single_continuum*u.Jy, spectral_axis=x_single_continuum*u.um)
+
+    # Smooth in the same way fit_generic_continuum does.
+    spectrum_smoothed = median_smooth(spectrum, 3)
+
+    # Window selects the first half of the spectrum.
+    g1_fit = fit_continuum(spectrum_smoothed, window=(0.*u.um, 5.*u.um))
+
+    spectrum_normalized = spectrum / g1_fit(spectrum.spectral_axis)
+
+    y_continuum_fitted_expected = np.ones(shape=(spectrum_normalized.spectral_axis.shape))
```

Comment:
Could we instead check against the results of a full fit?

```suggestion
   g1_fit_orig = fit_continuum(spectrum_smoothed)
   y_continuum_full = spectrum / g1_fit_orig(spectrum.spectral_axis)
   y_continuum_fitted_expected = y_continuum_full[(spectrum.spectral_axis >= 0 * u.um) & (spectrum.spectral_axis <= 5 * u.m)]
```

Reply:
When fitting over the entire spectrum, the fit gets very bad because of the narrow line at 6.3 um. So it doesn't seem reasonable to use that bad fit as a comparison reference. Unless we want to decrease the tolerance to values in the range of 0.2 or something like that.